clientconn.go

```diff
@@ -1032,6 +1032,7 @@ func (ac *addrConn) resetTransport() {
 				continue
 			}
 
+			backoffFor = 0
```

Comment:
Actually, this should also reset `backoffIdx = 0`.

Reply:
I have cleanup changes in another branch, was trying to keep this change simple so we can cherrypick it to previous releases.